src/backend/api/Fusion.Resources.Api/Deployment/k8s/deployment-pr-env.yml

```diff
@@ -0,0 +1,129 @@
+  labels: 
+    environment: pr
+    prNumber: '{{prNumber}}' #need quotes since it is an integer, k8s expects string
+type: Opaque
+stringData:
+  appsettings.secrets.yaml: |
+    {
+      "AzureAd": {
+        "ClientSecret": "{{clientSecret}}"
+      }
+    }
+---
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: 'resources-api-pr-{{prNumber}}'
+spec:
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 2
+      maxUnavailable: 50%
+  selector:
+    matchLabels:
+      app: 'resources-api-pr-{{prNumber}}'
+  template:
+    metadata:
+      labels:
+        app: 'resources-api-pr-{{prNumber}}'
+    spec:
+      volumes:
+      - name: env-secrets-volumes
+        secret: 
+          secretName: env-secrets
+
+      containers:
+      - name: resources-api-pr-{{prNumber}}
+        image: {{fusionAcr}}/{{imageName}}
+        ports:
+        - containerPort: 80
+        volumeMounts:
+        - name: env-secrets-volumes
+          mountPath: "/app/secrets" 
+        env:
+        - name: APPINSIGHTS_INSTRUMENTATIONKEY
+          value: "{{instrumentationKey}}"
+        - name: KEYVAULT_URL
+          value: "{{envKeyVaultUrl}}"
+        - name: ENVNAME
+          value: "{{ENVNAME}}"
+        - name: FUSION_ENVIRONMENT
+          value: "{{fusionEnvironment}}"
+        - name: AzureAd__ClientId
+          value: "{{clientId}}"
+        - name: ConnectionStrings__ResourcesDbContext
+          value: "{{dbConnectionString}}"
+        resources:
+          requests:
+            memory: "250Mi"
+            cpu: "50m"
+          limits:
+            memory: "500Mi"
+            cpu: "200m"
+        args:
+        - /server
+
+        livenessProbe:
+          httpGet:
+            path: /_health/liveness
+            port: 80
+          failureThreshold: 10
+          timeoutSeconds: 30
+          initialDelaySeconds: 60
+          periodSeconds: 30
+
+        readinessProbe:
+          httpGet:
+            path: /_health/ready
+            port: 80
+          initialDelaySeconds: 10
+          timeoutSeconds: 30
+          periodSeconds: 30
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: resources-api-pr-{{prNumber}}
+spec:
+  selector:
+    app: resources-api-pr-{{prNumber}}
+  ports:
+  - port: 80
+    targetPort: 80
+
+---
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: resources-api-fusion-ingress-{{prNumber}}
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    nginx.ingress.kubernetes.io/rewrite-target: /
+    nginx.ingress.kubernetes.io/proxy-buffer-size: "32k"
+    nginx.org/client-max-body-size: "50m"
+    nginx.ingress.kubernetes.io/client-max-body-size: "50m"
+    nginx.ingress.kubernetes.io/proxy-body-size: "50m"
+    fusion-health/environment: 'pr'
+    fusion-health/application: 'Fusion Resources'
+    fusion-health/url: /_health/ready
+spec:
+  tls:
+  - hosts:
+    - resources-api-pr-{{prNumber}}.fusion-dev.net
+    secretName: pr.fusion-dev.net-tls
```

Comment:
Maybe add a comment at the top of the yaml file that explains that this secret is required to exist?

Reply:
Good suggestion.